lib/auth.js

```diff
@@ -118,8 +118,10 @@ module.exports = {
       app_token: appToken
     };
 
-    this._authenticate(requestData, function(responseData) {
-      this._onAccessTokenAcquired(responseData, callback);
+    this._authenticate(requestData, function(err, responseData) {
+      if (err === null) {
+        this._onAccessTokenAcquired(responseData, callback);
+      }
```

Comment:
You're right, was assuming that _authenticate will call the callback itself, but we don't pass it the callback. Will fix